Create user_security schema

kimlisa opened the pull request:
task: https://github.com/hwsc-org/hwsc-user-svc/issues/38
- organized user_svc and user_security schema into its own folders
- changed naming of user_security schema files
- tested user_security schema

![screen shot 2019-02-11 at 1 22 58 am](https://user-images.githubusercontent.com/43280172/52554410-98e78000-2d9b-11e9-8562-38f5200c0269.png)

kimlisa committed:
Create user_security schema

- organized user_svc and user_security schema into its own folders
- changed naming of user_security schema files
- tested user_security schema

faraonc commented on int/hwsc-user-svc/schema/user-security/0_initial_user_sec_schema.up.sql:
```diff
@@ -0,0 +1,42 @@
+CREATE TYPE user_security.permission_level AS ENUM
+(
+  'NO_PERM',
+  'USER',
```
I need a permission called `USER_REGISTRATION`

kimlisa committed:
Make Permission and Ulid data type Global

- restructured folder structure
- combined user_svc and user_security schema under one script

faraonc commented on int/hwsc-user-svc/psql/0_initial_schemas.up.sql:
```diff
@@ -0,0 +1,96 @@
+  'NO_PERM',
+  'USER',
+  'ADMIN',
+  'USER_REGISTRATION'
```
I prefer USER_REGISTRATION as the second element or index 1, because I will be comparing the int value. the higher the int value the higher the access level. this makes it easy to do the following:
```
if token.permission < requiredPermision {
            return "UNAUTHENTICATED
}
```

kimlisa committed:
Switch order of permission enum values

-add newline to each sql file

faraonc commented on int/hwsc-user-svc/psql_test_insert.sql:
```diff
@@ -28,4 +28,15 @@ UPDATE user_svc.accounts SET
+INSERT INTO user_security.tokens (token_string, secret_key, token_type, algorithm, permission, expiration_date, uuid)
+VALUES
+  ('some token', 'someSecretKey', 'JWT', 'HS256', 'USER', current_timestamp, '0000xsnjg0mqjhbf4qx1efd6y3'),
+  ('another token', 'anotherSecretKey', 'NO_TYPE', 'NO_ALG', 'NO_PERM', current_timestamp, '0000xsnjg0mqjhbf4qx1efd6y6');
```
needs a new line

faraonc commented on int/hwsc-user-svc/psql/0_initial_schemas.up.sql:
```diff
@@ -0,0 +1,96 @@
+  permission        permission_level NOT NULL,
+  expiration_date   TIMESTAMPTZ NOT NULL,
+  uuid              ulid NOT NULL UNIQUE
+);
```
new line

faraonc commented on int/hwsc-user-svc/psql/0_initial_schemas.down.sql:
```diff
@@ -0,0 +1,2 @@
+DROP SCHEMA user_svc CASCADE;
+DROP SCHEMA user_security CASCADE;
```
newline

faraonc reviewed (approved):
TU, https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file
